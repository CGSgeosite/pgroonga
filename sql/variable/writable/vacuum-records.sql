CREATE TABLE memos (
  content text
);

CREATE INDEX pgrn_index ON memos USING PGroonga (content);

INSERT INTO memos VALUES ('Groonga is fast!');
DELETE FROM memos;

SET pgroonga.writable = false;

VACUUM memos;
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;

SET pgroonga.writable = true;

INSERT INTO memos VALUES ('PGroonga is fast!');
DELETE FROM memos;

VACUUM memos;
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;

DROP TABLE memos;
