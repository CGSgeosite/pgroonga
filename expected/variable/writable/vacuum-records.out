CREATE TABLE memos (
  content text
);
CREATE INDEX pgrn_index ON memos USING PGroonga (content);
INSERT INTO memos VALUES ('Groonga is fast!');
DELETE FROM memos;
SET pgroonga.writable = false;
VACUUM memos;
ERROR:  pgroonga: can't delete bulk records while pgroonga.writable is false
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;
                                    ?column?                                    
--------------------------------------------------------------------------------
 [[[1], [["_id", "UInt32"], ["content", "LongText"]], [1, "Groonga is fast!"]]]
(1 row)

SET pgroonga.writable = true;
INSERT INTO memos VALUES ('PGroonga is fast!');
DELETE FROM memos;
VACUUM memos;
SELECT pgroonga_command('select',
			ARRAY[
			  'table', pgroonga_table_name('pgrn_index'),
			  'output_columns', '_id, content'
			])::jsonb->>1;
                       ?column?                        
-------------------------------------------------------
 [[[0], [["_id", "UInt32"], ["content", "LongText"]]]]
(1 row)

DROP TABLE memos;
